src/modules/flight_mode_manager/tasks/Auto/FlightTaskAuto.cpp

```diff
@@ -266,23 +266,36 @@ void FlightTaskAuto::_prepareLandSetpoints()
 		Vector2f sticks_ne = sticks_xy;
 		Sticks::rotateIntoHeadingFrameXY(sticks_ne, _yaw, _land_heading);
 
-		const float distance_to_circle = math::trajectory::getMaxDistanceToCircle(_position.xy(), _initial_land_position.xy(),
-						 _param_mpc_land_radius.get(), sticks_ne);
+		const bool land_radius_enabled = _param_mpc_land_radius.get() > 0.0f;
+		const bool position_valid = !isnan(_position(0)) && !isnan(_position(1));
+
 		float max_speed;
```

Comment:
```suggestion
		float max_speed = INFINITY;
```
It's a limit applied on top of the default `MPC_VEL_MANUAL` speed it goes with full stick input.